sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -812,6 +812,75 @@ class MapboxMap internal constructor(
     }
   }
 
+  /**
+   * Returns all the leaves (original points) of a cluster (given its cluster_id) from a GeoJsonSource, with pagination support: limit is the number of leaves
+   * to return (set to Infinity for all points), and offset is the amount of points to skip (for pagination).
+   *
+   * Requires configuring the source as a cluster by calling [GeoJsonSource.Builder#cluster(boolean)].
+   *
+   * @param sourceIdentifier GeoJsonSource identifier.
+   * @param cluster Cluster from which to retrieve leaves from
+   * @param limit The number of points to return from the query (must use type 'uint64_t', set to maximum for all points). Defaults to 10.
+   * @param offset The amount of points to skip (for pagination, must use type 'uint64_t'). Defaults to 0.
+   * @param callback The result will be returned through the [QueryFeatureExtensionCallback].
+   *         The result is a feature collection or a string describing an error if the operation was not successful.
+   */
+  @JvmOverloads
+  fun getGeoJsonClusterLeaves(
+    sourceIdentifier: String,
+    cluster: Feature,
+    limit: Long = QFE_DEFAULT_LIMIT,
+    offset: Long = QFE_DEFAULT_OFFSET,
+    callback: QueryFeatureExtensionCallback,
+  ) {
+    if (limit < 0 || offset < 0) {
```

Comment:
if `limit` will be `0` what is the purpose of such call?

Reply:
Also - we probably could use kotlin unsigned types for such cases? 
Though they're still in Beta since 1.3, stable only since 1.5

Reply:
@yunikkk Thanks for your advice, but we will not add such checks in this pr. Worth to considerate this if we decide to add checks later.